LibraBFT/Impl/Consensus/Liveness/ProposerElection.agda

```diff
@@ -30,4 +30,4 @@ isValidProposalM b =
 
 isValidProposerM a r = isValidProposer <$> use lProposerElection <*> pure a <*> pure r
 
-isValidProposer pe a r = ⌊ getValidProposer pe r ≟ℕ a ⌋
+isValidProposer pe a r = toBool $ getValidProposer pe r == a
```

Comment:
Very minor nit, but the `toBool $` reveals that the `==` isn't quite equivalent to `==` in Haskell.  Given that, is this any better than  keeping `⌊ ... ⌋`?  I'm not saying you should make it so the code can look identical, unless you feel the urge and can do it without too much pain.

Reply:
The most lightweight change would be to change the type of `isValidProposer`, so I'll do that.

Reply:
Scratch my last comment -- I've changed it so that `_==_` returns `Bool`, as it does in Haskell.